packages/list/index.js

```diff
@@ -82,219 +85,188 @@ export default class List extends Component {
-        this.setState({listItemClassList});
-      },
-      focusItemAtIndex: (index) => {
-        const id = this.getListItemIdFromIndex_(index);
-        const listItem = this.listItems_[id];
-        if (listItem) {
-          listItem.focus();
+        const {listItemClassNames} = this.state;
+        if (!listItemClassNames[index]) {
+          return;
         }
+        const i = listItemClassNames[index].indexOf(className);
+        listItemClassNames[index].splice(i, 1);
+        this.setState({listItemClassNames});
       },
       setTabIndexForListItemChildren: (listItemIndex, tabIndexValue) => {
         const {listItemChildrenTabIndex} = this.state;
-        const id = this.getListItemIdFromIndex_(listItemIndex);
-        listItemChildrenTabIndex[id]= tabIndexValue;
+        listItemChildrenTabIndex[listItemIndex] = tabIndexValue;
         this.setState({listItemChildrenTabIndex});
       },
+      focusItemAtIndex: (index) => {
+        this.setState({focusListItemAtIndex: index});
+      },
       followHref: (index) => {
-        const id = this.getListItemIdFromIndex_(index);
-        const listItem = this.listItems_[id];
-        if (listItem) {
-          listItem.followHref();
-        }
+        this.setState({followHrefAtIndex: index});
       },
       toggleCheckbox: (index) => {
-        const id = this.getListItemIdFromIndex_(index);
-        const listItem = this.listItems_[id];
-        if (listItem) {
-          listItem.toggleCheckbox();
-        }
+        this.setState({toggleCheckboxAtIndex: index});
       },
     };
   }
 
-  getListItemIdFromIndex_ = (index) => {
-    for (let id in this.listItemIndices_) {
-      if (this.listItemIndices_[id] === index) return id;
-    };
-    return '';
-  }
+  render() {
+    const {
+      /* eslint-disable no-unused-vars */
+      className,
+      nonInteractive,
+      dense,
+      avatarList,
+      twoLine,
+      singleSelection,
+      selectedIndex,
+      wrapFocus,
+      /* eslint-enable no-unused-vars */
+      children,
+      ...otherProps
+    } = this.props;
 
-  getIndexOfListItemElement_ = (element) => {
-    let listItemId = '';
-    for (let id in this.listItems_) {
-      if (this.listItems_[id].listItemElement === element) {
-        listItemId = id;
-        break;
-      }
-    }
-    return this.listItemIndices_[listItemId];
+    this.listItemCount = 0;
+    return (
+      <ul
+        className={this.classes}
+        {...otherProps}
+      >
+        {React.Children.map(children, this.renderChild)}
+      </ul>
+    );
   }
 
-  getListItemIndexOfTarget_ = (eventTarget) => {
-    let target = eventTarget;
-
-    // Find the first ancestor that is a list item.
-    while (this.getIndexOfListItemElement_(target) === undefined) {
-      if (!target || target === document) {
-        return -1;
-      }
-      target = target.parentElement;
+  renderChild = (child) => {
+    if (child.type.name === 'ListItem') {
```

Comment:
I think in the Readme, we should say that the children need to be type ListItem's (if we don't already). One of the issues this past week gave an example of this:

```
<Link>
  <ListItem ... />
</Link>
```